src/unstract/sdk/vector_db.py

```diff
@@ -1,32 +1,59 @@
 import logging
-from typing import Union
+from collections.abc import Sequence
+from typing import Any, Optional, Union
 
-from llama_index.core.vector_stores.types import BasePydanticVectorStore, VectorStore
+from llama_index.core import StorageContext, VectorStoreIndex
+from llama_index.core.indices.base import IndexType
+from llama_index.core.schema import BaseNode, Document
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    VectorStore,
+    VectorStoreQueryResult,
+)
+from typing_extensions import deprecated
 from unstract.adapters.constants import Common
 from unstract.adapters.vectordb import adapters
 from unstract.adapters.vectordb.constants import VectorDbConstants
 
 from unstract.sdk.adapters import ToolAdapter
 from unstract.sdk.constants import LogLevel, ToolEnv
-from unstract.sdk.exceptions import SdkError, ToolVectorDBError
+from unstract.sdk.embedding import Embedding
+from unstract.sdk.exceptions import SdkError, VectorDBError
 from unstract.sdk.platform import PlatformHelper
 from unstract.sdk.tool.base import BaseTool
 
 logger = logging.getLogger(__name__)
 
 
-class ToolVectorDB:
+class VectorDB:
     """Class to handle VectorDB for Unstract Tools."""
 
-    def __init__(self, tool: BaseTool):
-        self.tool = tool
-        self.vector_db_adapters = adapters
+    vector_db_adapters = adapters
+    DEFAULT_EMBEDDING_DIMENSION = 1536
 
-    def __get_org_id(self) -> str:
+    def __init__(
+        self,
+        tool: BaseTool,
+        adapter_instance_id: str,
+        embedding: Optional[Embedding] = None,
+    ):
+        self._tool = tool
+        self._adapter_instance_id = adapter_instance_id
+        if embedding:
+            self._embedding_instance = embedding._embedding_instance
+            self._embedding_dimension = embedding._length
+        else:
+            self._embedding_dimension = VectorDB.DEFAULT_EMBEDDING_DIMENSION
+
+        self._vector_db_instance: Union[
+            BasePydanticVectorStore, VectorStore
+        ] = self._get_vector_db()
+
+    def _get_org_id(self) -> str:
         platform_helper = PlatformHelper(
-            tool=self.tool,
-            platform_host=self.tool.get_env_or_die(ToolEnv.PLATFORM_HOST),
-            platform_port=int(self.tool.get_env_or_die(ToolEnv.PLATFORM_PORT)),
+            tool=self._tool,
+            platform_host=self._tool.get_env_or_die(ToolEnv.PLATFORM_HOST),
+            platform_port=self._tool.get_env_or_die(ToolEnv.PLATFORM_PORT),
```

Comment:
@gaya3-zipstack Later we need to see how to retrieve platform key automatically without a tool env. Reason being sdk invocations can happen from outside tools too.